### PR TITLE
feat: add docker support to action directly

### LINKED ISSUE #374: Add docker support for GitHub action
#### Describe the bug
- Pulling an image from docker will not work on action.
- The image has to be added in action.yml for proper functioning.

#### TO-DO
- [x] Update README to use official names as per marketplace instead of docker-image
- [x] Add support for a bump of version in action.yml version.

### PR DESCRIPTION
**Pull request by Shurtu-gal**
**Description**

- Update README to use official names as per marketplace instead of docker-image
- Add support for a bump of version in action.yml version.

**Related issue(s)**
Fixes #374, #368

### CODE REVIEW

**Comment by Shurtu-gal** [comment]
Successful workflow run :- https://github.com/ash17290/test-experiment/actions/runs/7464972818/job/20313136124

As can be seen with the new one docker image is directly pulled saving precious time.

**Comment by Shurtu-gal** [comment]
Successful release run :- https://github.com/ash17290/github-action-for-cli/actions/runs/7466719118/job/20318704391

File: README.md [hunk 1]
```diff
@@ -91,15 +91,20 @@ The command that you use might support and even require specific parameters to b
...
+> However, you don't need to worry as it won't build the image every time. It will pull it from Docker Hub as it is already built there.
+
 ### Basic
 
 In case all defaults are fine for you, just add such step:
 
 ```yaml
 - name: Generating Markdown from my AsyncAPI document
-  uses: docker://asyncapi/github-action-for-cli:3.0.0
+  uses:
```
Comment (derberg) [inline README.md:107]: empty?

File: .releaserc [hunk 2]
```diff
@@ -18,4 +18,10 @@ plugins:
   - preset: conventionalcommits
 - - "@semantic-release/release-notes-generator"
   - preset: conventionalcommits
-- "@semantic-release/github"
+- - "@semantic-release/exec"
+  - prepareCmd: "VERSION=${nextRelease.version} npm run bump:version"
+- - "@semantic-release/git"
+  - assets:
+    - "action.yml"
+    message: "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}"
```
Comment (derberg) [inline .releaserc:26]: is this going to do a commit or PR? cause commit will not work as commits directly to master are blocked

File: Makefile [hunk 3]
```diff
@@ -14,6 +14,12 @@ export GITHUB_WORKSPACE = $(shell pwd)
 run:
 	@bash ./entrypoint.sh $(DEFAULT_VERSION) $(DEFAULT_COMMAND) $(TEST_FILEPATH) $(DEFAULT_TEMPLATE) $(DEFAULT_LANGUAGE) $(DEFAULT_OUTPUT) $(DEFAULT_PARAMETERS) $(DEFAULT_CUSTOM_COMMANDS) 
 
+# Updates the action.yml file with the latest version of the action got from $VERSION env variable
+generate-assets:
+	version=$$(echo ${VERSION} | sed 's/v//g'); \
+	sed "s/docker:\/\/asyncapi\/github-action-for-cli:.*/docker:\/\/asyncapi\/github-action-for-cli:$$version'/g" action.yml > action.yml.tmp
+	@mv action.yml.tmp action.yml
```
Comment (derberg) [inline Makefile:21]: it is a bit hack to run bump script as part of tests

can't we run it through npm scripts maybe?

Comment (Shurtu-gal) [reply Makefile:21]: No these are not run as part of tests. All commands in make-file are separate script.

File: README.md (hunk 1 above)
Comment (Shurtu-gal) [reply README.md:107]: My bad

File: .releaserc (hunk 2 above)
Comment (Shurtu-gal) [reply .releaserc:26]: It does a direct commit as making a PR won't work if we want to include it in release.

Comment (derberg) [reply .releaserc:26]: so can be removed

**Comment by derberg** [comment]
ping me when you need another review

**Comment by Shurtu-gal** [comment]
@derberg It is ready for review now

**Review by derberg — APPROVED** [review]

**Pull request merged**